datafusion/functions/src/unicode/character_length.rs

```diff
@@ -92,15 +81,32 @@ impl ScalarUDFImpl for CharacterLengthFunc {
 /// Returns number of characters in the string.
 /// character_length('josé') = 4
 /// The implementation counts UTF-8 code points to count the number of characters
-fn character_length<T: ArrowPrimitiveType>(args: &[ArrayRef]) -> Result<ArrayRef>
+fn character_length(args: &[ArrayRef]) -> Result<ArrayRef> {
+    match args[0].data_type() {
+        DataType::Utf8 => {
+            let string_array = args[0].as_string::<i32>();
+            character_length_general::<Int32Type, _>(string_array)
+        }
+        DataType::LargeUtf8 => {
+            let string_array = args[0].as_string::<i64>();
+            character_length_general::<Int64Type, _>(string_array)
+        }
+        DataType::Utf8View => {
+            let string_array = args[0].as_string_view();
+            character_length_general::<Int32Type, _>(string_array)
+        }
+        _ => unreachable!(),
+    }
+}
+
+fn character_length_general<'a, T: ArrowPrimitiveType, V: ArrayAccessor<Item = &'a str>>(
```

Comment:
This pattern comes up a lot (specifically using `ArrayAccessor<Item = &'a str>`) and I think will be one of the key pattens to implement native / specialized support for StringView across the codebase

I wonder if there is any way to make this more discoverable / approachable 🤔 

I am thinking I should adapt this code as a documentation example to the `ArrayAccessor` trait https://docs.rs/arrow/latest/arrow/array/trait.ArrayAccessor.html